### PR TITLE
Warn about automatic alignment of GeoSeries

### PR DESCRIPTION
**Pull request by gahjelle**
Starts addressing #750 

Gives a warning when doing operations on two GeoSeries that are not aligned to each other.

Currently the warning is quite terse. We should also add a page in the documentation that explains how the user can do what he/she really wants to do (reindexing, spatial join, etc), and then reference that page in the warning.

### CODE REVIEW

File: geopandas/tests/test_geoseries.py [hunk 1]
```diff
@@ -102,6 +103,14 @@ def test_align(self):
         assert a1['B'].equals(a2['B'])
         assert a1['C'].is_empty
 
+    def test_warning_if_not_aligned(self):
+        # Test that warning is issued when operating on non-aligned series
+        with pytest.warns(UserWarning):
+            self.a1.contains(self.a2)
+
+        with pytest.warns(UserWarning):
+            self.a1.union(self.a2)
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_geoseries.py]: Can you also add a test to explicitly test that no warning is raised in the case the index is equal?

Comment (gahjelle) [reply geopandas/tests/test_geoseries.py]: Done :blush:

File: geopandas/base.py [hunk 2]
```diff
@@ -41,7 +41,11 @@ def _delegate_binary_method(op, this, other, *args, **kwargs):
     # type: (str, GeoSeries, GeoSeries) -> GeoSeries/Series
     this = this.geometry
     if isinstance(other, GeoPandasBase):
-        this, other = this.align(other.geometry)
+        if not this.index.equals(other.index):
+            warn('The index of the two GeoSeries are different')
```
Comment (martinfleis) [inline geopandas/base.py]:
```suggestion
            warn('The indices of the two GeoSeries are different.')
```
Minor language-based suggestion.

Comment (gahjelle) [reply geopandas/base.py]: Thanks @martinfleis 

I've updated the text to use "indices" and the trailing dot.

**Comment by martinfleis** [comment]
Thank you @gahjelle and @jorisvandenbossche!

**Pull request merged**